src/lib.rs

```diff
@@ -91,12 +91,20 @@ pub fn auto_inherit() -> Result<(), anyhow::Error> {
         .expect("Failed to find `[workspace.dependencies]` table in root manifest.");
     let mut was_modified = false;
     for (package_name, source) in &package_name2inherited_source {
-        insert_preserving_decor(
-            workspace_deps,
-            package_name,
-            dep2toml_item(&shared2dep(source)),
-        );
-        was_modified = true;
+        if workspace_deps.get(package_name).is_some() {
+            eprintln!(
+                "`{}` won't be auto-inherited because it's already present in the root manifest.",
+                package_name
+            );
```

Comment:
```suggestion
```

Reply:
If we arrive here, it means that the workspace dependency for that package had a version compatible with other instances of the same package across the workspace.  
Therefore we don't need to do anything, but it is incorrect to say that it won't be auto-inherited, because it will be.